### DISCUSSION TITLE
Why is getting a street address so difficult?

### DISCUSSION THREAD

**Discussion by DetroitLibertyPenguin** [osmandapp/OsmAnd#15158] [General]
When I do a POI search the locations given doubt seem to give a steer address (I. E. Search  for Starbucks) more over if I just type q street address in it never finds anything. OCCASIONALLY I find since field that let's me enter an address one element at a time, wherever that is it must be barried somewhere because I can't usually find it. When I do it expects the information in like a backwards way, like state, then city, than street than house number. Or something. Is this normal in other parts of the world and it's just not yet optimized for USA?

**Comment by vshcherb** [comment]
We use as a reference https://nominatim.openstreetmap.org/ui/search.html. If it works there, we try to fix in our app as well. Also we try to update documentation how it works internally: https://docs.osmand.net/docs/technical/algorithms/trace-address-search-issues.

**Comment by DetroitLibertyPenguin** [comment]
So no simple address in standard American format works like "3059
Cumberland Rd., Berkley, MI" What is the best way for me to request sucH.
Even doing the structured search by town first does not seem to work well.
It feels like the system does not expect their to be multiple towns with
the same name. In the U. S. many, if not most, municipalities have the same
name as those in other states. Occasionally you can get the same name in
the same state (there are over 100 municipalities in the US called
"Springfield" for example.

On Monday, August 29, 2022, vshcherb ***@***.***> wrote:

> We use as a reference https://nominatim.openstreetmap.org/ui/search.html.
> If it works there, we try to fix in our app as well. Also we try to update
> documentation how it works internally: https://docs.osmand.net/docs/
> technical/algorithms/trace-address-search-issues.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/osmandapp/OsmAnd/discussions/15158#discussioncomment-3500868>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APSNIJ6S7A4FR6A74GCNN2TV3TBFPANCNFSM57VIBSCQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Sent from my SmartPhone please dismiss not-so-smart typos

**Reply by jfburke3** [reply]
I'd try spelling out Road to begin with.

**Reply by DetroitLibertyPenguin** [reply]
In case you are unaware using full names like Road, Street , Avenue , Boulevard, Highway , etc.  Is extremely rare in my country.

For example the official adress set by the government is:

3059 Cumberland Rd.
Berkley, MI 48072

People do say Road when we speak it allowed, but its not actually accurate

**Reply by DetroitLibertyPenguin** [reply]
Using the String "Road" Instead of "Rd." did not give me signifly better results. I tried other adresses as recommended. The app seems to not properly process adresses in American format.

**Reply by vshcherb** [reply]
As I see:
```3059 Cumberland Rd. Berkley, MI 48072``` -> Street found at 3rd place
```3059 Cumberland Berkley ```  -> Street at 1st place

Please not there is no house found cause it's not on the map

**Reply by Joeviocoe** [reply]
Pretty much unusable with American formats.  
This needs an Online/Internet search feature to translate.  Perhaps using duckduckgo.
There needs to be a way I can paste in a full address, as I'm not going to type in each field.

**Comment by DetroitLibertyPenguin** [comment]
but a street isn't an address, right?

Jim Fulner
22100 Armada Ridge Rd.,
Armada, MI 48005-3204
(P) 248-971-0259
(M) 248-765-7458
(H) 586-473-7007

Telephony eliminates confusion

**Comment by DetroitLibertyPenguin** [comment]
and agian this isn't about one location, but any "one shot addresss"

Jim Fulner
22100 Armada Ridge Rd.,
Armada, MI 48005-3204
(P) 248-971-0259
(M) 248-765-7458
(H) 586-473-7007

Telephony eliminates confusion

**Reply by tommycrock** [reply]
It's not clear what you expect, that adding a person's name and phone number to a search will help find results? Are you aware that osmand is based on the data in OSM and those details will not be recorded? Also house numbers will not be present unless someone like you has added them. Where someone has, searches of the kind you suggested before work. Completely random example, 
843 Niagara St. Elmont, NY 11003
Finds it as the top hit
![Screenshot_20230117_235418_OsmAnd+](https://user-images.githubusercontent.com/5749884/213037783-b4549484-17fe-4683-bf1c-ab4f3a877f60.jpg)
By the way, the structured search works the way it does to narrow down the search (like starting with a country and a postcode) but it would be nice to also be able to work the other way.
